A4A: add forms for new signup flow

andrii-lysenko opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Resolves https://github.com/Automattic/automattic-for-agencies-dev/issues/1728

## Proposed Changes

Adds pages to new signup flow

https://github.com/user-attachments/assets/8b2acb13-827c-4ba0-8f74-24ea2b5795f5


**Screenshots for translations:**

<img width="806" alt="Screenshot 2025-02-05 at 8 42 04 PM" src="https://github.com/user-attachments/assets/1984b1ed-31c5-4263-9d3c-70a040c491d7" />

<img width="796" alt="Screenshot 2025-02-05 at 8 42 10 PM" src="https://github.com/user-attachments/assets/436bafdb-09d1-401a-b68d-0ea99676358c" />

<img width="710" alt="Screenshot 2025-02-05 at 8 42 15 PM" src="https://github.com/user-attachments/assets/23103f32-fce2-4dfe-9d78-731c27911f9c" />

<img width="764" alt="Screenshot 2025-02-05 at 8 42 20 PM" src="https://github.com/user-attachments/assets/c5312d17-93b9-4ee8-83e6-00bcfdd8fb35" />

<img width="685" alt="Screenshot 2025-02-05 at 8 42 25 PM" src="https://github.com/user-attachments/assets/885e572d-dfb5-4fb7-96ad-a370c5cb7a9b" />


## Why are these changes being made?
<!--
It's easy to see what a PR does but much harder to find out why it was made,
particularly when researching old changes in history. Record an explanation of
the motivation behind this change and how it will help.
-->


## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

- Navigate to `/signup/wc-asia` and check behavior

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

andrii-lysenko committed:
A4A: add signup multiple steps form

andrii-lysenko committed:
A4A: add forms for new signup flow

yashwin commented on client/a8c-for-agencies/sections/signup/signup-v2/components/multi-step-form/blueprint-form/index.tsx:
```diff
@@ -0,0 +1,180 @@
+import FormField from 'calypso/a8c-for-agencies/components/form/field';
+import FormRadio from 'calypso/components/forms/form-radio';
+import FormTextarea from 'calypso/components/forms/form-textarea';
+import './style.scss';
```
Can we add an extra line before `import './style.scss';`. We have this throughout the app.


```suggestion

import './style.scss';
```

yashwin commented on client/a8c-for-agencies/sections/signup/signup-v2/components/multi-step-form/blueprint-form/index.tsx:
```diff
@@ -0,0 +1,180 @@
+					isRequired
+				>
+					<div className="blueprint-form__radio-group">
+						<FormRadio
+							label={ translate( 'Access to cutting-edge technology' ) }
+							checked={ formData.topGoal === 'cutting_edge_tech' }
+							onChange={ () => setFormData( { ...formData, topGoal: 'cutting_edge_tech' } ) }
+						/>
+						<FormRadio
+							label={ translate( 'Comprehensive support and troubleshooting' ) }
+							checked={ formData.topGoal === 'comprehensive_support' }
+							onChange={ () => setFormData( { ...formData, topGoal: 'comprehensive_support' } ) }
+						/>
+						<FormRadio
+							label={ translate( 'Lead generation and new business opportunities' ) }
+							checked={ formData.topGoal === 'lead_generation' }
+							onChange={ () => setFormData( { ...formData, topGoal: 'lead_generation' } ) }
+						/>
+						<FormRadio
+							label={ translate( 'Training and education for your team' ) }
+							checked={ formData.topGoal === 'training_education' }
+							onChange={ () => setFormData( { ...formData, topGoal: 'training_education' } ) }
+						/>
+						<FormRadio
+							label={ translate( 'Exclusive discounts or revenue-sharing options' ) }
+							checked={ formData.topGoal === 'exclusive_discounts' }
+							onChange={ () => setFormData( { ...formData, topGoal: 'exclusive_discounts' } ) }
+						/>
```
[Non-blocker] Maybe we could thinking of creating an array with all the radio fields and map through it and render instead of repeating it.

andrii-lysenko replied:
Sure, think it's great idea. This is just a quickly bootstrapped version so I could start testing out the backend endpoints. I think we could address these in specific tasks for each page

yashwin commented on client/a8c-for-agencies/sections/signup/signup-v2/components/multi-step-form/blueprint-form/index.tsx:
```diff
@@ -0,0 +1,180 @@
+					isRequired
+				>
+					<div className="blueprint-form__radio-group">
+						<FormRadio
+							label={ translate( 'Access to cutting-edge technology' ) }
+							checked={ formData.topGoal === 'cutting_edge_tech' }
+							onChange={ () => setFormData( { ...formData, topGoal: 'cutting_edge_tech' } ) }
+						/>
```
Can we add `htmlFor` and `id` to all the fields to make the text clickable to check/uncheck the radio.

```suggestion
						<FormRadio
							htmlFor="cutting_edge_tech"
							id="cutting_edge_tech"
							label={ translate( 'Access to cutting-edge technology' ) }
							checked={ formData.topGoal === 'cutting_edge_tech' }
							onChange={ () => setFormData( { ...formData, topGoal: 'cutting_edge_tech' } ) }
						/>
```

yashwin commented on client/a8c-for-agencies/sections/signup/signup-v2/components/multi-step-form/blueprint-form/style.scss:
```diff
@@ -0,0 +1,5 @@
+.blueprint-form {
+    h1 {
+        text-wrap: pretty;
```
This isn't supported on Firefox & Safari: https://developer.mozilla.org/en-US/docs/Web/CSS/text-wrap#browser_compatibility

We could use preventWidows on the element directly instead.

import { preventWidows } from 'calypso/lib/formatting';

yashwin commented on client/a8c-for-agencies/sections/signup/signup-v2/components/multi-step-form/choice-blueprint/index.tsx:
```diff
@@ -0,0 +1,36 @@
+			className="choice-blueprint-form"
+			title={ translate( 'Grow your business with a free personalized blueprint' ) }
+			description={ translate(
+				'By answering a few simple questions, we’ll provide tips on maximizing your agency’s impact.'
```
This is not an apostrophe sign:

<img width="757" alt="Screenshot 2025-02-03 at 1 24 34 PM" src="https://github.com/user-attachments/assets/4fb048bb-3651-4a0e-8fe1-857c0e3e946a" />

```suggestion
				"By answering a few simple questions, we'll provide tips on maximizing your agency's impact."
```

andrii-lysenko replied:
Good catch. These are just copied from Figma.

yashwin commented on client/a8c-for-agencies/sections/signup/signup-v2/components/multi-step-form/choice-blueprint/style.scss:
```diff
@@ -0,0 +1,22 @@
+.choice-blueprint-form {
+    h1 {
+        text-wrap: pretty;
```
Same as above, let's avoid using `text-wrap: pretty`

yashwin commented on client/a8c-for-agencies/sections/signup/signup-v2/components/multi-step-form/choice-blueprint/style.scss:
```diff
@@ -0,0 +1,22 @@
+}
+
+.choice-blueprint__text {
+    font-size: 1rem;
+    font-weight: 400;
```
Can we use [A4A Typography](https://github.com/Automattic/wp-calypso/blob/a5453358292e766c2064549e7987d5dfcd7a14db/client/assets/stylesheets/shared/mixins/_a4a-typography.scss) here? This will make it easy for us to migrate to Core Typography.

yashwin commented on client/a8c-for-agencies/sections/signup/signup-v2/components/multi-step-form/choice-blueprint/style.scss:
```diff
@@ -0,0 +1,22 @@
+.choice-blueprint__text {
+    font-size: 1rem;
+    font-weight: 400;
+    color: var(--studio-gray-60);
```
It is better to avoid variables from the studio everywhere on this file, and instead, we should use `--color-neutral`. This will ensure we are using the color based on the theme.

yashwin commented on client/a8c-for-agencies/sections/signup/signup-v2/components/multi-step-form/finish-signup-survey/index.tsx:
```diff
@@ -0,0 +1,33 @@
+
+	return (
+		<Form
+			className="finish-signup-survey-form"
```
Are we planning to use this `className` in the future? Looks like it is unused. Let's remove if that's not the case

yashwin commented on client/a8c-for-agencies/sections/signup/signup-v2/components/multi-step-form/finish-signup-survey/index.tsx:
```diff
@@ -0,0 +1,33 @@
+		>
+			<div>
+				<Button
+					className="finish-signup-survey-form__button"
```
Same as above. Let's remove if that's not the case

yashwin commented on client/a8c-for-agencies/sections/signup/signup-v2/components/multi-step-form/index.tsx:
```diff
@@ -24,9 +31,20 @@ const MultiStepForm = () => {
+			case 5:
+				return <FinishSignupSurvey onContinue={ () => setCurrentStep( 6 ) } />;
+			case 6:
+				return <h1>Thank you!</h1>;
```
Should we translate this?

andrii-lysenko replied:
It's a placeholder for now. We will check if we would need this page or not.

yashwin commented on client/a8c-for-agencies/sections/signup/signup-v2/components/multi-step-form/personalization/index.tsx:
We need to add htmlFor for all the fields to make them accessible

yashwin commented on client/a8c-for-agencies/sections/signup/signup-v2/components/multi-step-form/personalization/style.scss:
```diff
@@ -0,0 +1,91 @@
+    .a4a-form__section-field {
+        margin-bottom: 0;
+
+        &-label {
```
Can we avoid using `&-` in the file? This will make it a lot easier to search.

yashwin commented on client/a8c-for-agencies/sections/signup/signup-v2/components/multi-step-form/personalization/style.scss:
```diff
@@ -0,0 +1,91 @@
+        &-label {
+            font-size: rem(14px);
+            font-weight: 600;
+            color: var(--studio-gray-80);
```
It is better to avoid variables from the studio everywhere on this file, and instead, we should use --color-neutral. This will ensure we are using the color based on the theme.

yashwin commented on client/a8c-for-agencies/sections/signup/signup-v2/components/multi-step-form/personalization/style.scss:
```diff
@@ -0,0 +1,91 @@
+        margin-bottom: 0;
+
+        &-label {
+            font-size: rem(14px);
+            font-weight: 600;
```
Can we use A4A Typography.

yashwin reviewed (commented):
Thanks for working on this!

I have left some comments. Please let me know your thoughts!

jkguidaven commented:
For scroll behavior we should probably move the scrollbar at the edge.

<img width="1695" alt="Screenshot 2025-02-03 at 9 34 17 PM" src="https://github.com/user-attachments/assets/aedf598b-9694-4bf8-8f8a-96024f0d0be6" />

andrii-lysenko committed:
Refactor per suggestions

andrii-lysenko committed:
use preventWindows

yashwin reviewed (approved):
@andrii-lysenko: It looks like you left a [comment](https://github.com/Automattic/wp-calypso/pull/99137#issuecomment-2632367160) on a different PR.

I'm okay with addressing the comments in a different PR as long as we keep track of it. They are very easy to miss, as they are in 2 different PRs. 

I would suggest we address the comments related to styles in this PR. But, I'll leave that to you! 

I'm approving the PR now, as these changes are not live yet. 

I think we should also add a feature flag for this as the page is live in production.

andrii-lysenko committed:
adding feature flag and refactor styles

andrii-lysenko commented:
@yashwin yes, sorry for confusion. So I did some updates:
- Closed https://github.com/Automattic/wp-calypso/pull/99137 to avoid confusion and merge problems.
- Added feature flag `a4a-wc-asia-signup-enabled` so new page is only available when it's enabled.
- Fixed most small issues except: `Typography` usage, `htmlFor` and `id` used for form fields, and mapping fields as you suggested. These can be addressed in separate PRs for each page.

I will go ahead and merge this one so we could move forward with tasks for each page separately. 
The main goal of this PR was to have a quick prototype ready to be able to test backend flows earlier and address all the problems and styles separately, which come out not that ideal.

cc @jkguidaven

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/17231082

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Hi @andrii-lysenko, could you please edit the description of this PR and add a screenshot for our translators? Ideally it'd include all of the following strings:
- `Let's build your blurprint`
- `We'll send you a custom blueprint to grow your business based on your answers below.`
- `What is your top goal when partnering with a technology provider?`
- `Access to cutting-edge technology`
- `Comprehensive support and troubleshooting`
- `Lead generation and new business opportunities`
- `Training and education for your team`
- `Exclusive discounts or revenue-sharing options`
- `What is the main goal you hope to achieve in 2025?`
- `Scaling the agency significantly`
- `Diversifying offered services`
- `Increasing the number of long-term clients`
- `Expanding into new markets or territories`
- `Strengthening the agency's brand and reputation`
- `How does your agency typically work with clients regarding Automattic's solutions?`
- `We refer customers to Automattic products/services to purchase on their own`
- `We purchase on our clients' behalf and resell Automattic products/services`
- `A combination of referring and reselling`
- `Other models (please explain)`
- `Add your explanation`
- `Is there anything specific about your agency's approach or any challenges you face that you would like us to consider when creating your blueprint?`
- `Add your approach`
- `Grow your business with a free personalized blueprint`
- `By answering a few simple questions, we'll provide tips on maximizing your agency's impact.`
- `Sign up and unlock the blueprint to grow your agency's business`
- `Join 5000+ agencies and grow your business with Automattic for Agencies.`
- `We have sent you an email with more details about the program and instructions for logging in. You will also receive your blueprint in the coming days; keep an eye out for it!`
- `Close survey`
- `Finish survey`
- `Personalize your experience`
- `We'll tailor the product and onboarding for you.`
- `Where is your agency located?`
- `Select country`
- `Select option`
- `Developer at an agency`
- `Sales, marketing, or operations at an agency`
- `Freelancer`
- `50-100`
- `What Automattic products do you currently offer your clients?`

Thank you in advance!

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/17231082).
